src/vivarium_census_prl_synth_pop/tools/make_results.py

```diff
@@ -548,3 +552,67 @@ def subset_results_by_state(processed_results_dir: str, state: str) -> None:
             )
             write_to_disk(state_data, output_file_path)
         logger.info(f"Finished writing {observer} files for {state_name}.")
+
+
+def write_shard_metadata(
+    observer: str, obs_data: pd.DataFrame, obs_dir: Path, seed: str
+) -> None:
+    # Writes metadata for each shard of data. This will be proportions available to noise
+    # for specific columns.
+
+    def _get_metadata_values(df: pd.DataFrame, location: str, year: int) -> pd.DataFrame:
+        metadata_df = pd.DataFrame(index=[location])
+        metadata_df["number_of_rows"] = len(df)
+        metadata_df["year"] = year
+        copy_columns = [
+            col.name
+            for col in COLUMNS
+            for noise_type in col.noise_types
+            if "copy_from_household_member" == noise_type.name
+        ]
+        nicknames_columns = [
+            col.name
+            for col in COLUMNS
+            for noise_type in col.noise_types
+            if "use_nickname" == noise_type.name
+        ]
+        for column in df.columns:
+            if column in copy_columns:
+                # Get number of rows that could potentially copy a household member
+                metadata_df[f"{column}.copy_from_household_member"] = (
+                    df[COPY_HOUSEHOLD_MEMBER_COLS[column]].notna().sum()
+                )
+            elif column in nicknames_columns:
+                # Get number of rows eligible to be noised to a nickname
+                nicknames = load_nicknames_data()
+                metadata_df[f"{column}.use_nickname"] = df[column].isin(nicknames.index).sum()
+            else:
+                # TODO: Add guardian based duplication
+                continue
```

Comment:
Instead of doing the above, you can actually be even cleaner by doing this:
```python
for column_name in df.columns:
    column = COLUMNS.get_column(column_name)
    if "copy_from_household_member" in column.noise_types:
        ...
    if "use_nickname" in column.noise_types:
        ...
```

Reply:
The logic isn't that simple because column.noise_types is a tuple of noise types and each noise type has different attributes that's why I'm getting the different list of columns upfront.